gapis/trace/android/mali/profiling_data.go

```diff
@@ -164,3 +166,96 @@ func processCounters(ctx context.Context, processor *perfetto.Processor, desc *d
 	}
 	return counters, nil
 }
+
+const (
+	summaryCounterGroup   uint32 = 1
+	defaultCounterGroup   uint32 = 2
+	textureCounterGroup   uint32 = 3
+	loadStoreCounterGroup uint32 = 4
+	memoryCounterGroup    uint32 = 5
+	cacheCounterGroup     uint32 = 6
+)
+
+func updateCounterGroups(ctx context.Context, data *profile.ProfilingData) {
+	data.CounterGroups = append(data.CounterGroups,
+		&service.ProfilingData_CounterGroup{
+			Id:    summaryCounterGroup,
+			Label: "Performance",
+		},
+		&service.ProfilingData_CounterGroup{
+			Id:    defaultCounterGroup,
+			Label: "Overview",
+		},
+		&service.ProfilingData_CounterGroup{
+			Id:    textureCounterGroup,
+			Label: "Texture",
+		},
+		&service.ProfilingData_CounterGroup{
+			Id:    loadStoreCounterGroup,
+			Label: "Load/Store",
+		},
+		&service.ProfilingData_CounterGroup{
+			Id:    memoryCounterGroup,
+			Label: "Memory",
+		},
+		&service.ProfilingData_CounterGroup{
+			Id:    cacheCounterGroup,
+			Label: "Cache",
+		},
+	)
+	summaryCounters := map[string]struct{}{
+		"gpu time":                                   struct{}{},
```

Comment:
Do they really need to be `struct{}{}` ? Or can we have some sort of type here?

`struct{}{}` does not give any information about what actually are these values , how they calculated, what are their representations? If I would be looking at this to do something, I am completely lost.

Reply:
This is Go's way to do sets: https://www.davidkaya.com/sets-in-golang/
We don't care what the value of an entry in the map is, we just care about whether it is present or not. Since we don't care, the empty type is the best type to use. If we used something like a boolean, it'd be ambiguous what an entry's value of `true` means compared to `false`.